### ISSUE TITLE
Data handling from external API, parameter setting, filtering

### ISSUE THREAD

**Issue by mattijn** [vega/vega#820]
We've build dataservice and depending on the parameter setting it gives us a JSON object with different values (template is same).
This service follows the syntax:
`http://service.com/xyz?database=test&parameter_A=1&parameter_B=4&parameter_C=3`

This works well in data format like this 
```
  "data": [
    {
      "name": "points",
      "url": "http://service.com/xyz?database=test&parameter_A=1&parameter_B=4&parameter_C=3",
      "format": {
        "type": "json",
        "parse": {
          "x": "number",
          "y": "number",
          "kruinhoogte": "number"
        }
      }
   ]
```
like this we can manually adapt the parameters in the URL. 

But if I try the following to connect the URL from a signal object, but this seems not supported (as it won't do anything)

```
  ...
  "signals": [{
  	"name": "externalAPI",
  	"update": "http://service.com/xyz?database=test&parameter_A=1&parameter_B=4&parameter_C=3"
  }],
  "data": [{
  	"name": "points",
  	"url": {
  		"signal": "externalAPI"
  	},
  	...
  }]
```

Eventually I would like to request the external dataservice like the [crossfilter ](https://vega.github.io/vega/examples/crossfilter-flights/)example with multiple input ranges. 

Currently in application format we do it using javascript and requesting the data beforehand and load it into a `data | values` object and do no adopt a `data | url` object. But this is not our preferred route.

Do I miss something or this is not supported. Any advice would be very much appreciated.

**Comment by tnavda** [comment]
mattijn,

I don't believe this is possible at a first glance.  I thought it might be doable using a signal binding and the documentation suggests that any HTML input can be used, i.e. url.  However, searching through the code there is a switch statement that only supports the aforementioned HTML elements.

> Required. The type of input element to use. The valid values are checkbox, radio, range, select, and any other legal HTML form input type.

But the code below didn't work or make a call for that matter.  I will dig around some more through the code.
`
{ "name": "w", "value": "http://localhost/test.json",
"bind": {"input": "url" } },`

**Comment by lukasheinrich** [comment]
hi @mattijn ,

you can currently have a similar effect by setting up signal listeners (addSignalListener) to catch updates on signal values and then in the callback use the view API to update the data

something like
```
view.addSignalListener(signal, function(name, value) {
var signal_list = ["parameter_A","parameter_B","parameter_C"];
var all_signals = {}
$.each(signal_list,function(i,d){
					return all_signals[d] = view.signal(d)
				})
//now you call to get new data depending on values in all_signals
var new_data = $.get(...)

				view.change('points',vega.changeset().remove(view.data('points'))).run()
				view.change('points',vega.changeset().insert(new_data)).run()
})
```

**Comment by mattijn** [comment]
@lukasheinrich, thanks for sharing. I was hoping for a vega only approach. It's  more modular and makes the vega spec easier to share

**Comment by jheer** [comment]
Currently, we do not have a "pure Vega" approach for _dynamic_ data loading (e.g., parameterized by signals), nor for polling from data sources that update over time (as in #893). For now, the solution is to use the JavaScript View API to marshall data and issue change sets. We're open to data loading extensions in the future (probably not until version 3.1 or later), but these will require extensions to the specification language design as well as internal logic.

**Comment by mattijn** [comment]
Added in [Vega 4.2](https://github.com/vega/vega/releases/tag/v4.2.0)
